src/ruby/lib/grpc/generic/client_stub.rb

```diff
@@ -334,17 +338,18 @@ def server_streamer(method, req, marshal, unmarshal,
       if return_op
         # return the operation view of the active_call; define #execute
         # as a new method for this instance that invokes #server_streamer
-        c.merge_metadata_to_send(metadata)
```

Comment:
This is the call that is causing the tests to fail. The metadata is sent before the interceptors have a chance to run and add additional metadata values.